events/events.go

```diff
@@ -0,0 +1,197 @@
+package events
+
+import (
+	"errors"
+	"sync"
+)
+
+var errClosed = errors.New("publisher is closed")
+
+// GetFn returns the event
+type GetFn func() interface{}
+
+// Event is an event
+type Event interface {
+	// OnGet sets the function to be executed when Get() is called
+	OnGet(GetFn)
```

Comment:
I was thinking about cases where some subscribers are only interested on knowing the updates but not the data, thus made the GetFn to be optional for those subscribers, but I guess they can just keep the GetFn but not call the Get() if they are not interested in the data